docs/tutorials/custom_segmentation_trainer.ipynb

```diff
@@ -0,0 +1,514 @@
+    "                ),\n",
+    "                'MeanIoU': JaccardIndex(\n",
+    "                    num_classes=num_classes, task='multiclass', average='macro'\n",
+    "                ),\n",
+    "            },\n",
+    "            prefix='train_',\n",
+    "        )\n",
+    "        self.val_metrics = self.train_metrics.clone(prefix='val_')\n",
+    "        self.test_metrics = self.train_metrics.clone(prefix='test_')\n",
+    "\n",
+    "    def configure_callbacks(self) -> Sequence[Callback] | Callback:\n",
+    "        \"\"\"Initialize callbacks for saving the best and latest models.\n",
+    "\n",
+    "        Returns:\n",
+    "            List of callbacks to apply.\n",
+    "        \"\"\"\n",
+    "        return [\n",
+    "            ModelCheckpoint(every_n_epochs=50, save_top_k=-1, save_last=True),\n",
+    "            ModelCheckpoint(monitor=self.monitor, mode=self.mode, save_top_k=5),\n",
+    "        ]\n",
+    "\n",
+    "    def on_train_epoch_start(self) -> None:\n",
+    "        \"\"\"Log the learning rate at the start of each training epoch.\"\"\"\n",
+    "        optimizers = self.optimizers()\n",
+    "        if isinstance(optimizers, list):\n",
+    "            lr = optimizers[0].param_groups[0]['lr']\n",
+    "        else:\n",
+    "            lr = optimizers.param_groups[0]['lr']\n",
+    "        self.logger.experiment.add_scalar('lr', lr, self.current_epoch)  # type: ignore"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Train model\n",
+    "\n",
+    "The remainder of the turial is straightforward and follows the typical [PyTorch Lightning](https://lightning.ai/) training routine. We instantiate a `DataModule` for the LandCover.AI 100 dataset (a small version of the LandCover.AI dataset for notebook testing), instantiate a `CustomSemanticSegmentationTask` with a U-Net and ResNet-50 backbone, then train the model using a Lightning trainer."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dm = LandCoverAI100DataModule(root='data', batch_size=10, num_workers=2, download=True)\n",
+    "\n",
+    "# You can use the following for actual training runs\n",
+    "# from torchgeo.datamodules import LandCoverAIDataModule\n",
+    "# dm = LandCoverAIDataModule(root='data/', batch_size=64, num_workers=8, download=True)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "task = CustomSemanticSegmentationTask(\n",
+    "    model='unet',\n",
+    "    backbone='resnet50',\n",
+    "    weights=True,\n",
+    "    in_channels=3,\n",
+    "    num_classes=6,\n",
+    "    loss='ce',\n",
+    "    lr=1e-3,\n",
+    "    tmax=50,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
```

Comment:
I usually strip outputs from the notebooks before uploading them so the outputs don't change every time someone submits a PR. I know you prefer to include the output, and I'm trying to find ways to make both of us happy. Ideally, I want to store only `.py` files and generate the `.ipynb` on-the-fly but haven't figured out a way to do that yet.

Reply:
TorchVision, Kornia, Sklearn, Lightning, etc. all have notebook output. If you don't show the notebook output then you should just have .py files?

https://pytorch.org/vision/stable/auto_examples/transforms/plot_transforms_getting_started.html#sphx-glr-auto-examples-transforms-plot-transforms-getting-started-py

https://kornia.github.io/tutorials/nbs/image_matching_lightglue.html